common/dex-auth/base/dex/dex_configmap.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: dex
+data:
+  config.yaml: |
+    issuer: $(issuer)
+    storage:
+      type: kubernetes
+      config:
+        inCluster: true
+    web:
+      https: 0.0.0.0:5556
+      tlsCert: /etc/dex/tls/tls.crt
+      tlsKey: /etc/dex/tls/tls.key
+    logger:
+      level: "debug"
+      format: text
+
+    connectors:
```

Comment:
Should have a facility to be overlayed. This ensures different connectors can be able to connect to dex. LDAP is a reference implementation.

Reply:
Would a 'facility' be a tool?

Given your current layout
```
common/dex-auth/base
├── dex
│   ├── ca_configmap.yaml
│   ├── crd.yaml
│   ├── deployment.yaml
│   ├── dex_configmap.yaml
│   ├── kustomization.yaml
│   └── service.yaml
├── dex-k8s-authenticator
│   ├── configmap.yaml
│   ├── deployment.yaml
│   ├── kustomization.yaml
│   └── service.yaml
├── dex_ca_contents.pem
├── dex_params.env
├── dex_params.yaml
├── idp_ca_contents.pem
├── istio-auth
│   ├── authentication_policy.yaml
│   └── kustomization.yaml
├── k8s_ca_contents.pem
├── kustomization.yaml
└── ldap
    ├── README.md
    ├── admin_service.yaml
    ├── deployment.yaml
    ├── kustomization.yaml
    └── service.yaml
```

the subdirs: dex, dex-k8s-authenticator, istio-auth, ldap are bases referenced in dex/kustomization.yaml.
This is fine although bases go away in kustomize-v2.+ and just resources are used. The [BestPractices](https://github.com/kubeflow/manifests/blob/master/docs/KustomizeBestPractices.md) document suggests a directory structure along with how vars, overlays, configMaps etc should be used but if we abide by the document and current patterns in the other manifests your current layout would morph to:

```
.
├── dex-authn
│   └── base
│       ├── config-map.yaml
│       ├── deployment.yaml
│       ├── dex_ca_contents.pem
│       ├── idp_ca_contents.pem
│       ├── k8s_ca_contents.pem
│       ├── kustomization.yaml
│       ├── params.env
│       ├── params.yaml
│       └── service.yaml
├── dex-crds
│   └── base
│       ├── config-map.yaml
│       ├── crds.yaml
│       ├── deployment.yaml
│       ├── kustomization.yaml
│       └── service.yaml
└── dex-ldap
    └── base
        ├── deployment.yaml
        ├── kustomization.yaml
        └── service.yaml
```

What makes this PR challenging is that a number of new patterns are introduced 
1. tooling is embedded under dex-auth/certs. The admission-controller generates a cert using admission-webhook/bootstrap/base/config-map.yaml - you should do something similar. Or there is code in controller-runtime that creates a cert I think.
2. Istio related rules are inverted - rather than being under a particular manifest target like ./pipeline/pipelines-ui/overlays/istio, the manifest target is referenced out-of-context under ./dex-auth/authorization/Istio/.
3. common/dex-auth/authorization/Kubernetes looks like it holds ClusterRoles, ClusterRoleBindings where the subjects are groups: admin, validator, datascience
4. The ldap README.md references the groups mentioned in 3, but outlines a set of steps the admin needs to do to prior to using kfctl which defines a mapping of groups to RBAC. The problem is this isn't integrated with profiles or istio so i don't think it would work - at least not in gcp.  I'm guessing there isn't a kubeflow/bootstrap/config/*.yaml that kfctl uses where dex-auth is deployed and works with ldap. 

I think I need to go through the ldap use-case more carefully and deploy it via a config.file

Reply:
> Would a 'facility' be a tool?

Is there a way of overlaying a config from kfctl? Otherwise, we might need to introduce tooling around it. 

> Given your current layout

I'll adopt this structure according to v2.

> 
> What makes this PR challenging is that a number of new patterns are introduced
> 
>     1. tooling is embedded under dex-auth/certs. The admission-controller generates a cert using admission-webhook/bootstrap/base/config-map.yaml - you should do something similar. Or there is code in controller-runtime that creates a cert I think.

Will include this as an optional part of kfctl instead.

>     2. Istio related rules are inverted - rather than being under a particular manifest target like ./pipeline/pipelines-ui/overlays/istio, the manifest target is referenced out-of-context under ./dex-auth/authorization/Istio/.

I'll make this change.
 
>     3. common/dex-auth/authorization/Kubernetes looks like it holds ClusterRoles, ClusterRoleBindings where the subjects are groups: admin, validator, datascience

I'll move these to examples folder.

>     4. The ldap README.md references the groups mentioned in 3, but outlines a set of steps the admin needs to do to prior to using kfctl which defines a mapping of groups to RBAC. The problem is this isn't integrated with profiles or istio so i don't think it would work - at least not in gcp.  I'm guessing there isn't a kubeflow/bootstrap/config/*.yaml that kfctl uses where dex-auth is deployed and works with ldap.
> 
> 
> I think I need to go through the ldap use-case more carefully and deploy it via a config.file

LDAP instructions are just a guideline here to setup. I'll remove them from here and possibly add these to docs.